pkg/util/push/otel.go

```diff
@@ -150,10 +152,40 @@ func OTLPHandler(
 		)
 
 		req.Timeseries = metrics
+
+		metadata := otelMetricsToMetadata(otlpReq.Metrics())
+		req.Metadata = metadata
+
 		return body, nil
 	})
 }
 
+func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
+	var metadata []*mimirpb.MetricMetadata
+
+	resourceMetricsSlice := md.ResourceMetrics()
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		resourceMetrics := resourceMetricsSlice.At(i)
+		scopeMetricsSlice := resourceMetrics.ScopeMetrics()
+
+		for j := 0; j < scopeMetricsSlice.Len(); j++ {
+			scopeMetrics := scopeMetricsSlice.At(j)
+			for k := 0; k < scopeMetrics.Metrics().Len(); k++ {
+				entry := mimirpb.MetricMetadata{
+					Type:             mimirpb.MetricMetadata_MetricType(scopeMetrics.Metrics().At(k).Type()),
+					MetricFamilyName: prometheustranslator.BuildCompliantName(scopeMetrics.Metrics().At(k), "", true), // TODO expose addMetricSuffixes in configuration
```

Comment:
Is this TODO still relevant?

Reply:
Yes as per @gouthamve request, exposing this boolean into configuration should be done in separate PR
https://github.com/grafana/mimir/pull/5693#issuecomment-1669216300
Answering my question
https://github.com/grafana/mimir/issues/4910#issuecomment-1668452485